detect_secrets/plugins/keyword.py

```diff
@@ -50,109 +50,120 @@
-    'true):',
-    'true,',
-    'true;',
-    'undef',
-    'undef,',
-    '{',
-    '{{',
-}
-'''
+# Deprecated false positives list. This will be migrated soon.
+# FALSE_POSITIVES = {
+#     '""',
+#     '""):',
+#     '"\'',
+#     '")',
+#     '"dummy',
+#     '"replace',
+#     '"this',
+#     '#pass',
+#     '#password',
+#     '$(shell',
+#     "'\"",
+#     "''",
+#     "''):",
+#     "')",
+#     "'dummy",
+#     "'replace",
+#     "'this",
+#     '(nsstring',
+#     '-default}',
+#     '::',
+#     '<%=',
+#     '<?php',
+#     '<a',
+#     '<aws_secret_access_key>',
+#     '<input',
+#     '<password>',
+#     '<redacted>',
+#     '<secret',
+#     '>',
+#     '=',
+#     '\\"$(shell',
+#     '\\k.*"',
+#     "\\k.*'",
+#     '`cat',
+#     '`grep',
+#     '`sudo',
+#     'account_password',
+#     'api_key',
+#     'disable',
+#     'dummy_secret',
+#     'dummy_value',
+#     'false',
+#     'false):',
+#     'false,',
+#     'false;',
+#     'login_password',
+#     'none',
+#     'none,',
+#     'none}',
+#     'nopasswd',
+#     'not',
+#     'not_real_key',
+#     'null',
+#     'null,',
+#     'null.*"',
+#     "null.*'",
+#     'null;',
+#     'pass',
+#     'pass)',
+#     'password',
+#     'password)',
+#     'password))',
+#     'password,',
+#     'password},',
+#     'prompt',
+#     'redacted',
+#     'secret',
+#     'some_key',
+#     'str',
+#     'str_to_sign',
+#     'string',
+#     'string)',
+#     'string,',
+#     'string;',
+#     'string?',
+#     'string?)',
+#     'string}',
+#     'string}}',
+#     'test',
+#     'test-access-key',
+#     'thisisnottherealsecret',
+#     'todo',
+#     'true',
+#     'true):',
+#     'true,',
+#     'true;',
+#     'undef',
+#     'undef,',
+#     '{',
+#     '{{',
+# }
 # Includes ], ', " as closing
 CLOSING = r'[]\'"]{0,2}'
 DENYLIST_REGEX = r'|'.join(DENYLIST)
 # Non-greedy match
 OPTIONAL_WHITESPACE = r'\s*'
 OPTIONAL_NON_WHITESPACE = r'[^\s]{0,50}?'
-QUOTE = r'[\'"]'
-SECRET = r'[^\r\n]+'
+QUOTE = r'[\'"`]'
+# Secret regex details:
+#    [^\r\n]*      -> this section match with every character except line breaks.
+#                     This allows to find secrets that starts with symbols or
+#                     alphanumeric characters.
+#    [a-zA-Z0-9]+  -> this section match only with alphanumeric characters, and at
+#                     least one is required. This allows to reduce the false positives
+#                     number.
+#    [^\r\n]*      -> this section match with every character except line breaks.
+#                     This allows to find secrets with symbols at the end.
+#    [^\r\n,\'"`]  -> this section match with the last secret character that can be
+#                     everything except line breaks, comma, backticks or quotes. This
+#                     allows to reduce the false positives number and to prevent
+#                     errors in the code snippet highlighting.
+SECRET = r'[^\r\n]*[a-zA-Z0-9]+[^\r\n]*[^\r\n,\'"`]'
```

Comment:
@pablosantiagolopez : I think you might have introduced an accidental exponential performance degradation here, with files with long lines.

```
(stable_version) $ time detect-secrets scan detect_secrets/filters/gibberish/rfc.model
...
real    0m0.286s
user    0m0.243s
sys     0m0.036s

(version_on_master) $ time detect-secrets scan detect_secrets/filters/gibberish/rfc.model
...
real    29m28.225s
user    29m24.396s
sys     0m2.074s


(version_on_master) $ time detect-secrets scan detect_secrets/filters/gibberish/rfc.model --disable-plugin KeywordDetector
...
real    0m0.385s
user    0m0.322s
sys     0m0.054s
```

Please look into this.